aws/rust-runtime/aws-config/src/ecs.rs

```diff
@@ -700,23 +868,30 @@ mod test {
         );
         let err = validate_full_uri("http://www.amazon.com/creds", dns.clone())
             .await
-            .expect_err("not a loopback");
+            .expect_err("not a valid IP");
         assert!(
             matches!(
                 err,
                 InvalidFullUriError {
-                    kind: InvalidFullUriErrorKind::NotLoopback
+                    kind: InvalidFullUriErrorKind::DisallowedIP
                 }
             ),
             "{:?}",
             err
         );
-        assert!(logs_contain(
-            "Address does not resolve to the loopback interface"
-        ));
-        validate_full_uri("http://localhost:8888/creds", dns)
+        assert!(logs_contain("Address does not resolve to an allowed IP"));
+        validate_full_uri("http://localhost:8888/creds", dns.clone())
             .await
             .expect("localhost is the loopback interface");
+        validate_full_uri("http://169.254.170.2.backname.io:8888/creds", dns.clone())
```

Comment:
This definitely works, but I'm a little hesitant to have our tests rely on backname.io since it adds some level of risk to our release process if backname.io happens to go down for a period of time. Could you refactor the test to use a fake `ResolveDns` trait implementation instead of the real one?

Reply:
this test is skipped because it already relies on dns resolution (of amazon.com)
see comment `// ignored by default because it relies on actual DNS resolution`
Is there some release process that is unskipping it?

Reply:
Oh, nope. I just didn't see that comment. Looks good then.